### ISSUE TITLE
Visual glitches in UI

### ISSUE THREAD

**Issue by arnowelzel** [nextcloud/collectives#831]
**Describe the bug**
Certain areas in Collectives have visual glitches:

- New pages use a default symbol which can not be displayed.
- The selection background in "Recent pages" is too small.
- The header section for pages is not high enough, so lower parts of letters (for example lower part of "g") can get cut off.
- The emoji in the header section is inside a circle which cuts off the emoji itself if it is squared or needs more space then the circular area provides.
- The selected collective has a white text in the navigation instead of black.

**To Reproduce**
1. Open Collectives
2. Create a new collective
3. Create some sub pages and use the letter "g" and select a square emoji to see the header issues

**Expected behavior**
- New pages should have a default emoji for "page" since emojis can not be removed anyway once deleted.
- The grey selectoin box in "Recent pages" should be wide enough so when hovering a page, the page itself will not stick outside.
- The text of page headers should either be smaller, so it will always fit inside the header or the header needs to be a bit higher.
- The emoji button should not be circular but just a square with smaller rounded boxes (e.g. 4px border radius) so the emoji will not get cut off.

Note: I fixed some parts of that (except "Recent pages" which I did not investigate in detail yet)) for my own server using [CustomCSS](https://apps.nextcloud.com/apps/theming_customcss) and the following rules:

```
/* Collectives: fix emoji button - do not use round buttons which can cut off the emoji */

.app-collectives .button-emoji-page {
  border-radius: 4px !important;
}

/* Collectives: Adjust sizes of title headers to avoid descenders getting cut off */

.app-collectives input.title {
  height: 43px !important;
  font-size: 30px !important;
}

/* Collectives: fix text for selected collective in navigation */

.app-collectives .app-navigation-entry__title {
  color: var(--color-main-text) !important;
}
```

**Screenshots**

![image](https://github.com/nextcloud/collectives/assets/6613614/12f182c6-6398-426d-aab6-c8e4464a950d)

![image](https://github.com/nextcloud/collectives/assets/6613614/020a4aa2-f0fd-4376-8acc-30e8dfe12fe6)

**Client details:**
 - OS: Windows 10 x64 Pro
 - Browser: Vivaldi
 - Version: 6.1.3035.257
 - Device: desktop

Note: this is not OS or client dependent. Also other browsers and operating systems show this problem.

<details>
<summary>Server details</summary>

**Collectives app version:** 2.7.0

**Operating system:** Ubuntu Linux 22.04

**Web server:** Apache

**Database:** MariaDB

**PHP version:** 8.1

**Nextcloud version:** 27.0.2

</details>

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
```
(not relevant)
```

#### Browser log
```
(not relevant)
```

</details>

**Comment by mejo-** [comment]
Thanks for your feedback @arnowelzel.

> New pages use a default symbol which can not be displayed.

From your screenshots, it seems rather like the emoji chosen for your collective is not supported by your font. Could you please provide a screenshot with a page without a selected emoji?

> The selection background in "Recent pages" is too small.

Indeed, fix incoming.

> The header section for pages is not high enough, so lower parts of letters (for example lower part of "g") can get cut off.

This seems to be the case with some fonts. And indeed the font size was a bit too big for the input field. Fix incoming.

> The emoji in the header section is inside a circle which cuts off the emoji itself if it is squared or needs more space then the circular area provides.

Well, all our buttons have the same border radius. I fixed this by using a slightly smaller font size for the emoji instead. Fix incoming.

> The selected collective has a white text in the navigation instead of black.

Indeed, fix incoming.

**Comment by max-nextcloud** [comment]
> > The selected collective has a white text in the navigation instead of black.
> 
> Indeed, fix incoming.

For me the selected collective has a primary color background:

![grafik](https://github.com/nextcloud/collectives/assets/97337118/763e58e8-8d1a-4f00-a49c-be5c4881416c)

In this case white fonts would make sense. This is on Nextcloud 27.1.0

@arnowelzel did you change the primary color by chance?

**Comment by arnowelzel** [comment]
> > > The selected collective has a white text in the navigation instead of black.
> > 
> > 
> > Indeed, fix incoming.
> 
> For me the selected collective has a primary color background:
> 
> ![grafik](https://user-images.githubusercontent.com/97337118/269705009-763e58e8-8d1a-4f00-a49c-be5c4881416c.png)
> 
> In this case white fonts would make sense. This is on Nextcloud 27.1.0
> 
> @arnowelzel did you change the primary color by chance?

I tested that with Nextcloud 27.0 - maybe that already got fixed with 27.1.0 and the new Collectives app

**Comment by arnowelzel** [comment]
Indeed Collectives 2.7.1 changed the color scheme - however now "..." is black while the text is white:

![image](https://github.com/nextcloud/collectives/assets/6613614/e249e1fd-51e7-4ff7-99ab-0df41c9fac77)

**Comment by arnowelzel** [comment]
In addition "New collective" has a white "+" now:

![image](https://github.com/nextcloud/collectives/assets/6613614/98f29d19-9af0-4b6b-9c9a-0a29647cedae)

Seems the whole styling colors need some cleanup ;-)

**Comment by arnowelzel** [comment]
Some quick workarounds for Collectives 2.7.1 with CustomCSS:

```
.app-navigation-entry.active svg.material-design-icon__svg {
  fill: var(--color-primary-element-text);
}

.button-vue__icon:hover svg.material-design-icon__svg {
  fill: var(--color-main-text);
}
```

Result:

1) Selected item is now with white text and context menu icon in same color:

![image](https://github.com/nextcloud/collectives/assets/6613614/e0ea9f3d-8293-4997-a355-55231cd7d7c0)

2) Hovering the context menu or opening it will make the dots dark again (primary text color):

![image](https://github.com/nextcloud/collectives/assets/6613614/4fc48c81-d8be-4acc-8ade-8adb653f30ef)

![image](https://github.com/nextcloud/collectives/assets/6613614/97fea4b2-b550-49f7-bae5-7722d3d10264)

**Comment by mejo-** [comment]
> I wonder why Collectives can not just stick with the same color scheme which other apps like "Files", "Photos", "Activity" etc. use. Isn't there a global scheme for that kind of navigation?

That's actually what we try to do. We use the NcAppNavigationItem component from the `@nextcloud/vue` library for the collectives list. And the issue you faced with the bad contrast was a bug in this library that got fixed in the meantime. Wrong color of the three dots is also a bug in `@nextcloud/vue`. I opened https://github.com/nextcloud-libraries/nextcloud-vue/issues/4571 to track this.
